bittensor/_receptor/receptor_impl.py

```diff
@@ -123,20 +123,32 @@ def __del__ ( self ):
     def __exit__ ( self ):
         self.__del__()
 
-    def sign ( self ):
+    def sign_v1( self ):
         r""" Uses the wallet pubkey to sign a message containing the pubkey and the time
         """
-        nounce = self.nounce()
-        message  = str(nounce) + str(self.wallet.hotkey.ss58_address) + str(self.receptor_uid)
+        nonce = self.nonce()
+        message  = str(nonce) + str(self.wallet.hotkey.ss58_address) + str(self.receptor_uid)
         spliter = 'bitxx'
-        signature = spliter.join([ str(nounce), str(self.wallet.hotkey.ss58_address), "0x" + self.wallet.hotkey.sign(message).hex(), str(self.receptor_uid) ])
+        signature = spliter.join([ str(nonce), str(self.wallet.hotkey.ss58_address), "0x" + self.wallet.hotkey.sign(message).hex(), str(self.receptor_uid) ])
         return signature
-    
-    def nounce ( self ):
+
+    def sign_v2(self):
+        nonce = f"{self.nonce()}"
+        sender_hotkey = self.wallet.hotkey.ss58_address
+        receiver_hotkey = self.endpoint.hotkey
+        message = f"{nonce}.{sender_hotkey}.{receiver_hotkey}.{self.receptor_uid}"
+        signature = f"0x{self.wallet.hotkey.sign(message).hex()}"
+        return ".".join([nonce, sender_hotkey, signature, self.receptor_uid])
+
+    def sign(self):
+        if self.endpoint.version >= bittensor.__new_signature_version__:
```

Comment:
I like this design, really nice use of the versioning from the endpoint information. However, we have to make sure that the version information is updated properly on the chain.

Reply:
@camfairchild Do you know if the version information from the chain is updated based on every extrinsic or if it is updated based on the time of registration?

Reply:
Yea, the issue is that axon serving extrinsic is only submitted if the neuron IP or port has been changed. For older miners, the version information can be quite outdated. Instead, servers should submit a new extrinsic if the version information has also been changed

Reply:
https://github.com/opentensor/bittensor/pull/983/commits/0a91336fc8390570a10fc2ca03af5fd68df4825f should make the serve updates apply on every parameter change.

Reply:
I see, you are checking all of the neuron params. Nice change!